configtx/application.go

```diff
@@ -423,15 +431,359 @@ func (a *ApplicationOrg) setMSPConfig(updatedMSP MSP) error {
+func (m *ApplicationOrgMSP) RemoveAdminCert(cert *x509.Certificate) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	certs := msp.Admins[:]
+	for i, c := range msp.Admins {
+		if c.Equal(cert) {
+			certs = append(certs[:i], certs[i+1:]...)
+			break
+		}
+	}
+
+	msp.Admins = certs
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// AddRootCert adds an root certificate trusted by the organization MSP.
+func (m *ApplicationOrgMSP) AddRootCert(cert *x509.Certificate) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range msp.RootCerts {
+		if c.Equal(cert) {
+			return nil
+		}
+	}
+
+	msp.RootCerts = append(msp.RootCerts, cert)
+
+	err = msp.validateCACerts()
+	if err != nil {
+		return err
+	}
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// RemoveRootCert removes an trusted root certificate from the organization MSP.
+func (m *ApplicationOrgMSP) RemoveRootCert(cert *x509.Certificate) error {
+	msp, err := getMSPConfig(m.configGroup)
 	if err != nil {
-		return nil, fmt.Errorf("retrieving application org msp: %s", err)
+		return err
+	}
+
+	certs := msp.RootCerts[:]
+	for i, c := range msp.RootCerts {
+		if c.Equal(cert) {
+			certs = append(certs[:i], certs[i+1:]...)
+			break
+		}
+	}
+
+	msp.RootCerts = certs
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// AddIntermediateCert adds an intermediate certificate trusted by the organization MSP.
+func (m *ApplicationOrgMSP) AddIntermediateCert(cert *x509.Certificate) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range msp.IntermediateCerts {
+		if c.Equal(cert) {
+			return nil
+		}
+	}
+
+	msp.IntermediateCerts = append(msp.IntermediateCerts, cert)
+
+	err = msp.validateCACerts()
+	if err != nil {
+		return err
+	}
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// RemoveIntermediateCert removes an trusted intermediate certificate trusted by the organization MSP.
+func (m *ApplicationOrgMSP) RemoveIntermediateCert(cert *x509.Certificate) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	certs := msp.IntermediateCerts[:]
+	for i, c := range msp.IntermediateCerts {
+		if c.Equal(cert) {
+			certs = append(certs[:i], certs[i+1:]...)
+			break
+		}
+	}
+
+	msp.IntermediateCerts = certs
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// AddOUIdentifier adds a fabric organizational unit for the organization MSP.
+func (m *ApplicationOrgMSP) AddOUIdentifier(ou membership.OUIdentifier) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	for _, o := range msp.OrganizationalUnitIdentifiers {
+		if reflect.DeepEqual(o, ou) {
+			return nil
+		}
+	}
+
+	msp.OrganizationalUnitIdentifiers = append(msp.OrganizationalUnitIdentifiers, ou)
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// RemoveOUIdentifier removes a fabric organizational unit for the organization MSP.
```

Comment:
`RemoveOUIdentifier removes an existing organizational unit identifier from the organization MSP.`

Reply:
🛠️